### PR TITLE
Add deny dead_code lint and fix one error

### PR DESCRIPTION
**Pull request by bjfish**

### CODE REVIEW

**Comment by bjfish** [comment]
bors try

**Comment by bjfish** [comment]
bors try

**Comment by syrusakbary** [comment]
Merging directly since there are no conflicts and tests passed.

File: lib/emscripten/src/env/mod.rs [hunk 1]
```diff
@@ -30,6 +30,7 @@ pub fn call_malloc(ctx: &mut Ctx, size: u32) -> u32 {
         .unwrap()
 }
 
+#[warn(dead_code)]
```
Comment (MarkMcCaskey) [inline lib/emscripten/src/env/mod.rs]: is this different from `#[must_use]`?

Comment (MarkMcCaskey) [reply lib/emscripten/src/env/mod.rs]: Oh, it's normally deny but we're making it warn.  I understand

**Pull request merged**